### PR TITLE
Feature/evaluation metrics

### PR DESCRIPTION
**Pull request by fluegelk**
Adds computation of evaluation metrics from confusion matrices. The following metrics are added, their interfaces are based on the corresponding `sklearn.metrics` functions:
- **Accuracy:** the global accuracy
- **Balanced Accuracy:** the accuracy as average over class-wise recalls
- **Precision, Recall, and Fβ-Score:** with the following averaging options (`average` parameter)
  - `None`: no averaging, return class-wise
  - `"micro"`: compute metrics globally → equal importance on each sample
  - `"macro"`: compute metrics class-wise, then average over classes → equal importance on each class, minority classes can outweigh majority classes
  - `"weighted"`: compute metrics class-wise, then average over classes weighted by their support (#true samples)
  - As we are focussing on multi-class classification, the average options `"binary"` and `"samples"` are not included.
- **Cohen's Kappa:** compare classification to random guessing, values from -1 to +1, the higher the better, robust to class imbalance but not originally intended for classification problems
- **Matthews Correlation Coefficient (MCC):** see https://en.wikipedia.org/wiki/Phi_coefficient, values from -1 to +1, the higher the better, robust to class imbalance

Solves Issue #16

### CODE REVIEW

**Review by mcw92 — APPROVED** [review]
Looks good to me -- I made some small changes to the docstrings to appease the MyPy pre-commit hook. In addition, I removed MacOS from the test build matrix as one of the MacOS tests fails for dubious MPI reasons or just times out. The Python version differs among runs and I have no idea why that is right now. As this is not super important and I want the tests to work in the main branch, I will merge your branch now and work on implementing the confusion matrix.
